### ISSUE TITLE
`sentinelsat -d` on offline products sometimes triggers retrieval, sometimes crashes with code 500

### ISSUE THREAD

**Issue by daniel-j-h** [sentinelsat/sentinelsat#401]
Hey folks, I'm trying to download Sentinel-1 GRD data from 2019. It seems like these products are all / most offline by now and need to be retrieved from LTA. When I run the command line as in

    sentinelsat -g box.geojson --sentinel 1 --producttype GRD -s 20190305 -e 20190307 -d

I sometimes get the following logging messages - which indicate successful retrieval triggering

```
Found 1 products                                                                                         
Will download 1 products using 2 workers                                                                 
Product af2c3ab6-5094-47f7-b831-5656a16e4bef is in LTA.                                                
af2c3ab6-5094-47f7-b831-5656a16e4bef is not online.                                                      
af2c3ab6-5094-47f7-b831-5656a16e4bef accepted for retrieval      
```                                                                                                                                                  

but for some products I get a crash due to a 500 status code

```
Found 1 products
Will download 1 products using 2 workers
Product 6b9a9d37-1619-4daf-af8d-aabc2b1bdd91 is in LTA.
6b9a9d37-1619-4daf-af8d-aabc2b1bdd91 is not online.
Trying to download an offline product
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/sentinel/.local/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 837, in _trigger_offline_retrieval_until_stop
    status_code = self._trigger_offline_retrieval(product_info["url"])
  File "/home/sentinel/.local/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 666, in _trigger_offline_retrieval
    raise SentinelAPILTAError("Trying to download an offline product", r)
sentinelsat.sentinel.SentinelAPILTAError: HTTP status 500 Internal Server Error: Trying to download an offline product
Traceback (most recent call last):
  File "/home/sentinel/.local/bin/sentinelsat", line 8, in <module>
    sys.exit(cli())
  File "/home/sentinel/.local/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/sentinel/.local/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/sentinel/.local/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sentinel/.local/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/sentinel/.local/lib/python3.8/site-packages/sentinelsat/scripts/cli.py", line 228, in cli
    product_infos, triggered, failed_downloads = api.download_all(products, path)
  File "/home/sentinel/.local/lib/python3.8/site-packages/sentinelsat/sentinel.py", line 807, in download_all
    raise next(iter(x.exception() for x in dl_tasks if x.exception()))
StopIteration
```

I don't see a pattern in why I sometimes get a successful retrieval and sometimes not.

Why are we trying to download offline products in some cases, resulting in status code 500s from the api?

Related to https://github.com/sentinelsat/sentinelsat/issues/400 which is about success vs rate limit indications in the download library function.

**Comment by daniel-j-h** [comment]
Here is another Sentinel-1 product id I can trigger this crash / status code 500 with:

`def6dbab-c053-411a-bb0e-c5e1d58631dd`

reproducible with

    sentinelsat --uuid def6dbab-c053-411a-bb0e-c5e1d58631dd -d

**Comment by mirelsol** [comment]
Right now executing the query gives "accepted for retrieval".
I also have the same problem since a couple of weeks. This is a problem from the scihub server (unstable?).
However in my opinion the error is not handled correctly. In this case StopIteration shouldn't be raised.

**Comment by mirelsol** [comment]
> It causes a `StopIteration` if `dl_tasks` is empty.

It is probably the same issue as #377 and will be fixed in #378.

**Comment by valgur** [comment]
Should be fixed with #378. Feel free to re-open if that's not the case.
